songs/api/SongPermission.py

```diff
@@ -0,0 +1,21 @@
+from rest_framework import permissions
+
+
+class SongCreatePermission(permissions.BasePermission):
+    """
+    Permission to check if the user is admin or not
+    """
+
+    def has_permission(self, request, view):
+        if request.method in permissions.SAFE_METHODS:
+            return True
+        try:
+            if request.user.is_admin:
+                return True
+        except:
+            return False
+
+    def has_object_permission(self, request, view, obj):
```

Comment:
We do not allow deletion of songs?

Reply:
Yes we do allow. I have modified this

Reply:
test it

Reply:
if I request for specific song with id number then has_permission is called first and then has_object_permission gets called